conf_file.h

```diff
@@ -17,24 +17,123 @@
  */
 #pragma once
 
+#define DECLARE_PARSE_INT(_type) \
+    static int parse_##_type(const char *val, size_t val_len, void *storage, size_t storage_len)
```

Comment:
this is only useful for "internal" usage by the class. You don't need this defined for who includes this header. As such, a better approach would be to move the define inside the class and after it's used you undef it.